From 39d8316f2521b2810609ce7e8e2def8585ad05bc Mon Sep 17 00:00:00 2001
From: rhb2 <robert.bogart@joyent.com>
Date: Wed, 4 Sep 2019 18:57:12 +0000
Subject: [PATCH] MANTA-4531 Empty failed task list returned from agent appears
 as Some([]) instead of None Reviewed by: Rui Loura <rui.loura@joyent.com>
 Approved by: Rui Loura <rui.loura@joyent.com>

---
 src/agent.rs | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/agent.rs b/src/agent.rs
index cead4db..bc0a5f3 100644
--- a/src/agent.rs
+++ b/src/agent.rs
@@ -741,8 +741,13 @@ fn process_assignment(assignments: Arc<Mutex<Assignments>>, uuid: String) {
         tmp.tasks[i] = t;
     }
 
+    let failed = match failures.is_empty() {
+        true => None,
+        _ => Some(failures),
+    };
+
     assignment.write().unwrap().stats.state =
-        AgentAssignmentState::Complete(Some(failures));
+        AgentAssignmentState::Complete(failed);
     assignment_complete(assignments, uuid);
 }
 
-- 
2.21.0

